posts/comm-software-devel/index.qmd

```diff
@@ -0,0 +1,127 @@
+
+## Versioning
+
+Versioning code is not a new concept and has been used for decades[^1]. It has led to version control systems such as [git](https://git-scm.com/). However, in this post we are interested in versioning to communicate development.
+
+### Semantic versioning
+
+One such philosophy is [semantic versioning](https://semver.org/) (SemVer). This aims to describe the stage of software development by attaching semantics (i.e. meaning) to the format and numbering of versions. The version system works through three numbers, each separated by a dot. The numbers, from left to right, convey major version, minor version and patch version. As an example, 0.5.2, is newer than 0.3.9.
+
+Employing semantic versioning in ones code development allows others to determine whether a package has undergone substantial development and testing, and informs to whether it would make a suitable package to use in a script or as a dependency for another package. Semantic versioning also describes the changes made to a package. As explained on [their website](https://semver.org/), incrementing the major version implies a breaking change, a minor increment is a backwards compatible change and lastly patches are mostly applied to bug fixes. This aids users in understanding whether they should continue using a package, whether their package needs updating due to a breaking change or whether they need to install the newest version because a bug was recently fixed.
+
+In R there are several packages that work with versioning, and specifically semantic versioning. The {semver} package provides functions for parsing, rendering and comparing versions. There is also the {semverutils} R package which provides similar functionality using `R6`. The {usethis} package provides handy utility functions for changing the versions of R packages (`usethis::use_version()` and `usethis::use_dev_version()`). R also comes with a `package_version()` function for creating and validating versions.
+
+Overall semantic versioning provides what they describe as a "formal specification" to facilitate management of package development and the dependencies of that package. It is the most widely-used versioning system and therefore will be understood by a wide variety of users and developers.
+
+Some of the critique raised for semantic versioning is the difficulty of defining how changes correspond to a version increment. Semantic versioning states only breaking changes warrant major releases, but a vast re-write of a code base may also justify a major version change. Different breaking changes have different magnitudes, therefore a change to a single exported function or a change to every exported function will be communicated in a single, equal, version increment.
+
+### Alternatives to seminatic versioning
```

Comment:
```suggestion
### Alternatives to semiantic versioning
```

Reply:
Resolved in 6cd144edfa56aba433a5eff3a968d895b144bbe1